pasqal_cloud/__init__.py

```diff
@@ -151,11 +155,16 @@ def create_batch(
                 DeprecationWarning,
                 stacklevel=2,
             )
+            wait = wait or fetch_results
+
+        if wait and not complete:
+            raise ValueError("Cannot wait for an open batch.")
```

Comment:
In `add_jobs` the behaviour of wait is to wait until all jobs are in termination status. Here it is to wait until the batch is set to done.

Wouldn't it be more consistent to change the behaviour of the wait on open (and closed) batches to also depend on the status of all jobs?

Reply:
I updated it following your suggestion.